### PR TITLE
feat: [CDS-78627]: Added open api spec endpoint for runtime inputs API v1 pipeline yaml

### PR DESCRIPTION
**Pull request by mohitg0795**
## Describe your changes
Added new spec v1 endpoint for fetching details of runtime inputs in v1 pipeline yaml

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52393)

### CODE REVIEW

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52393)* status: 2 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2661 at r1](https://reviewable.io/reviews/harness/harness-core/52393#-NdTnnkbD3_TYhwSLMKA:-NdTnnkbD3_TYhwSLMKB:b-st6p6p) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L2661)):*
> ```YAML
>         id: t1e88zsr8os9p
>       enum:
>         - String
> ```

@mankrit-singh we are following small case for enums or first letter caps?

**Review by mankrit-singh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52393)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2661 at r1](https://reviewable.io/reviews/harness/harness-core/52393#-NdTnnkbD3_TYhwSLMKA:-NdTqHgI5ny_-4CVqeFU:b-e0brf6) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L2661)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

@mankrit-singh we are following small case for enums or first letter caps?
</blockquote></details>

We follow all caps generally, but we didn't define this as a standard I believe

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52393)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2661 at r1](https://reviewable.io/reviews/harness/harness-core/52393#-NdTnnkbD3_TYhwSLMKA:-NdTr7ztAhVFOem45VOl:b-byl62x) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L2661)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

We follow all caps generally, but we didn't define this as a standard I believe
</blockquote></details>




btw, I see multiple such examples already in the spec

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52393)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -769,15 +789,46 @@ paths:
...
         - $ref: '#/components/parameters/AccountHeader'
-      description: 'Pipeline Execution API'
+      description: Pipeline Execution API
       tags:
         - Pipeline Execution
+  '/v1/orgs/{org}/projects/{project}/pipelines/{pipeline}/inputs-schema':
+    parameters:
+      - schema:
+          type: string
+        name: org
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:799]: please use the common params for this, referring to other APIs

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 2]
```diff
@@ -2568,6 +2619,144 @@ components:
...
+        - Boolean
+        - Object
+      x-internal: true
+    PipelineYamlInputMetadataDTO:
+      title: PipelineYamlInputMetadataDTO
+      x-stoplight:
+        id: nzok6vvjmj4ww
+      type: object
+      properties:
+        fieldProperties:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:2672]: field_properties (casing)

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 3]
```diff
@@ -2568,6 +2619,144 @@ components:
...
+        dependencies:
+          $ref: '#/components/schemas/YamlInputDependencyDetailsDTO'
+      x-internal: true
+    InputDetailsPerFieldDTO:
+      title: InputDetailsPerFieldDTO
+      x-stoplight:
+        id: spzvqrltt78o4
+      type: object
+      properties:
+        inputType:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:2687]: input_type and internal_type

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 4]
```diff
@@ -2568,6 +2619,144 @@ components:
...
+          x-stoplight:
+            id: 6g2l9bcr1d5rt
+      x-internal: true
+    RuntimeInputDependencyDetailsDTO:
+      title: RuntimeInputDependencyDetailsDTO
+      x-stoplight:
+        id: psjgzoxyi1bij
+      type: object
+      properties:
+        fieldName:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:2702]: field_name, input_name

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 5]
```diff
@@ -2568,6 +2619,144 @@ components:
...
+          x-stoplight:
+            id: 846ox6eowvno0
+      x-internal: true
+    FixedValueFieldDependencyDetailsDTO:
+      title: FixedValueFieldDependencyDetailsDTO
+      x-stoplight:
+        id: 6zc1eytxqxgey
+      type: object
+      properties:
+        fieldName:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:2717]: same here

File: pipeline-service/contracts/openapi/v1/openapi.yaml [hunk 6]
```diff
@@ -2568,6 +2619,144 @@ components:
...
+          x-stoplight:
+            id: u1fn5c43ntyb5
+      x-internal: true
+    YamlInputDependencyDetailsDTO:
+      title: YamlInputDependencyDetailsDTO
+      x-stoplight:
+        id: zp85ezfip6v15
+      type: object
+      properties:
+        requiredRuntimeInputs:
```
Comment (mankrit-singh) [inline pipeline-service/contracts/openapi/v1/openapi.yaml:2736]: and here too

**Review by mankrit-singh — COMMENTED** [review]
same casing comments and use of common params

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52393)* status: 2 file(s) to review. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2687 at r1](https://reviewable.io/reviews/harness/harness-core/52393#gh-1314502859:-NdWjco53oSZS6MkkFKw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L2687)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

input_type and internal_type
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2702 at r1](https://reviewable.io/reviews/harness/harness-core/52393#gh-1314503061:-NdWjdOfAOo9KjqMCrRv:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L2702)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

field_name, input_name
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2717 at r1](https://reviewable.io/reviews/harness/harness-core/52393#gh-1314503211:-NdWjdkP7Ov0hsRoxzxm:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L2717)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

same here
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2736 at r1](https://reviewable.io/reviews/harness/harness-core/52393#gh-1314503580:-NdWje9o5yVVXYt-iHWi:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L2736)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

and here too
</blockquote></details>

Done.

**Comment by mohitg0795** [comment]
trigger uts

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52393)* status: 2 file(s) to review. 6. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 799 at r1](https://reviewable.io/reviews/harness/harness-core/52393#gh-1314501812:-Ndb8K864BCkcT0uJ1ht:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L799)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

please use the common params for this, referring to other APIs
</blockquote></details>

Done.
___
*[`pipeline-service/contracts/openapi/v1/openapi.yaml` line 2672 at r1](https://reviewable.io/reviews/harness/harness-core/52393#gh-1314502560:-Ndb7Dtb5qXf62MCzdVI:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/be7728d25d6f570f2184edb6a50016bf0d1cc28b/pipeline-service/contracts/openapi/v1/openapi.yaml#L2672)):*
<details><summary><i>Previously, mankrit-singh (Mankrit Singh) wrote…</i></summary><blockquote>

field_properties (casing)
</blockquote></details>

Done.

**Comment by mohitg0795** [comment]
trigger uts

**Comment by mohitg0795** [comment]
trigger uts

**Review by mankrit-singh — APPROVED** [review]

**Comment by mohitg0795** [comment]
trigger ut1

**Review by archit-harness — APPROVED** [review]

**Comment by mohitg0795** [comment]
trigger review

**Review by ankittiwari-harness — APPROVED** [review]

**Comment by mohitg0795** [comment]
trigger uts

**Pull request merged**